### DISCUSSION TITLE
Share a new trend I've just seen(about plot in datasette)

### DISCUSSION THREAD

**Discussion by HaveF** [rclement/datasette-dashboards#216] [General]
I read a newsletter by [SIMON WILLISON](https://substack.com/@simonw): [Stuff we figured out about AI in 2023](https://simonw.substack.com/p/stuff-we-figured-out-about-ai-in)

> Link 2023-12-31 [datasette-plot - a new Datasette Plugin for building data visualizations](https://www.datasette.cloud/blog/2023/datasette-plot/):

> I forgot to link to this here last week: Alex Garcia released the first version of datasette-plot, a brand new Datasette visualization plugin built on top of the Observable Plot charting library. We plan to use this as the new, updated alternative to my older datasette-vega plugin.

Although this dashboards plugin has nothing to do with their plugins, I still think it's interesting for you to see their conversion trends.

**Comment by rclement** [comment]
Thanks @HaveF for pointing-out to this news, I have been really busy lately and I do not manage to keep up with all the Datasette stuff coming in.

It is interesting to see some work towards more integrated data visualizations.

Let's hope Simon and co open the discussion and thinking about having these plot be part of a more global workflow so that it could be integrated with `datasette-dashboards`. For instance, I had plans on being able to create charts from tables interactively (as does `datasette-plot`) then adding them progressively to dashboards, but that would require a more complex paradigm where the charts and dashboards are stored within the database itself. Completely doable but that goes kind of against the initial value proposition of Datasette being bound to a static database and configuration, but I guess this is where the project is going now.

**Reply by HaveF** [reply]
> I have been really busy lately and I do not manage to keep up with all the Datasette stuff coming in.

I hope you are well. @rclement 

> I had plans on being able to create charts from tables interactively (as does datasette-plot) then adding them progressively to dashboards

To be honest, I think this is actually a bit difficult and need lots of work. The difficulty is that in fact most of the cases, the customized charts like datasette-plot or datasette-vega are a bit simple.

If we consider using more intelligent automatic recommendation charts, such as [vega/compassql: CompassQL Query Language for visualization recommendation.](https://github.com/vega/compassql) (vega ecosystem), this will make the project move in another direction (EDA).

The strategy I currently use is to use your dashboard to make a basic graphic. If I want more complicated use cases, I will consider using [vega/voyager: Visualization Tool for Data Exploration](https://github.com/vega/voyager) to explore other possible graphics and get basic vega lite spec. If I want to do more exquisitely, I will use Vega Online Editor to edit.

There are also some Vega ecosystem tools, such as [Kanaries/pygwalker: PyGWalker: Turn your pandas dataframe into an interactive UI for visual analysis](https://github.com/Kanaries/pygwalker), which can be explored data similar to Tableua. This tool is probably combined with Voyager(It has not been updated for a long time) and Vega. Maybe just wait is a good idea :-)

I totally agree with you a more global workflow should be useful. But there are some difficulties, such as personal preferences, etc.

Happy New Year!

**Comment by HaveF** [comment]
[datasette/studio: Launch Datasette Studio in a GitHub Codespace](https://github.com/datasette/studio)

maybe we can add dashboards to our demo in the future!
